tests/test_proxy_functional.py

```diff
@@ -3,13 +3,47 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def secure_proxy_url(tls_ca_certificate_pem_path, tls_certificate_private_key_path):
+    """Return the an instance of a running secure proxy."""
+    port = str(proxy.common.utils.get_available_port())
+    address = "127.0.0.1"
+    input_args = [
+        "--num-workers",
+        "1",
+        "--threadless",
+        "--hostname",
+        "127.0.0.1",
+        "--port",
+        port,
+        "--cert-file",
+        tls_ca_certificate_pem_path,
+        "--key-file",
+        tls_certificate_private_key_path,
+        "--threadless",
+    ]
+    proxy_url = f"https://{address}:{port}"
+    with proxy.Proxy(input_args=input_args):
+        yield proxy_url
+
+
+async def test_secure_proxy_http_absolute_path(secure_proxy_url, get_request) -> None:
```

Comment:
If you can, you could try sticking both HTTP and HTTPS into a single test. Just use `@pytest.mark.parametrize` for this.

Reply:
Will do.